addons/mail/models/models.py

```diff
@@ -399,10 +399,11 @@ def email_key(email):
         # make a record-based list of emails to give to '_partner_find_from_emails'
         records_emails = {}
         all_emails = set()
+        customers = {record._mail_get_customer().id for record in self}
```

Comment:
Meh ... not sure for this ... will have to think about it again. I understand the issue, but not sure it should be done here, as this method is used also in some other flows ... mehmeh.